### ISSUE TITLE
TIME as a substructure of DATE

### ISSUE THREAD

**Issue by hartenthaler** [FamilySearch/GEDCOM#236]
TIME is allowed in the standard as a substructure of DateValue. DateValue can be for example a DatePeriod:

> DateValue = date / DatePeriod / dateRange / dateApprox / ""

But what is the meaning of 
2 DATE FROM 1 JAN 2000 to 31 DEC 2000
3 TIME 12:34

btw: why is it "DatePeriod" with a capital "D" and not "datePeriod" like dateRange or dateApprox?

**Comment by fisharebest** [comment]
> But what is the meaning of

Suppose I told you that my child was born at 12:34, in the year 2000...

**Comment by tychonievich** [comment]
> btw: why is it "DatePeriod" with a capital "D" and not "datePeriod" like dateRange or dateApprox?

Per [their defining standard](https://www.rfc-editor.org/info/std68) ABNF rule names are case-insensitive. During the drafting of 7.0.0 we noticed that other standards seemed to use upper-case for externally visible rules and lower-case for internal-use-only rules, so we tried to follow that practice. Hence we used `DatePeriod` because it's used to define a datatype but `dateApprox` because it is only used to define part of `DateValue`.

That initial casing rule goal was difficult to formally define and follow. For example, `datePeriod` is used in the first paragraph of section 3.3.1; is that "externally visible" enough to deserve upper-case naming? We decided not, but also decided that this kind of corner case would make formal specification of the casing rules too difficult to define, which is why they are not called our explicitly in the specification.

**Comment by Norwegian-Sardines** [comment]
Actually FROM/TO or datePeriod means that an event happened continually FROM a TO b, so a TIME would be stating that at TIME c every day FROM a TO b the event occurred.  I would say that while logical, this type of event would rarely occur, maybe you could say: “The captains wife walked down to the dock to check if his ship arrived very day at noon from the day he left until she died”

The issue is that some illogical date/time combinations can be produced but sometimes a logical statement such as in “A child was born in 2012 at 14:32” needs to be recorded.

**Comment by tychonievich** [comment]
I agree with @hartenthaler that the specification does not define what it means to have a time under a variety of dates.

- Does `FROM <date> TO <date>` with `TIME <time>` mean the multi-day event both started and stopped at that time? Or was it a periodic event that took place each day at that time during that period?
- Does `BET <date> AND <date>` with `TIME <time>` mean we're not sure when it happened but we know it was between that time on those two dates? Or does it mean that we don't know the date, but we do know the time (I think this second option is what both @Norwegian-Sardines and @fisharebest are proposing in their comments)?
- Does `DATE 2022` with `TIME 11:50` mean 11:50 on some unknown day in 2022?
- Does `DATE ABT 24 OCT 2022` with `TIME 11:50 Z` mean we're unsure of the date but know the time? Or does it mean we are unsure of the exact time but know it was close to 2022-10-24 11:50:00 UTC?

Are there other cases we should consider?

I see several options for resolving this ambiguity of meaning

1. The spec could say "the meaning of TIME under a imprecise date is not defined by this specification."
2. The spec could define the meaning of each of these cases, perhaps with a general rule like "The time is always considered to be the precise time of the event. If the time is uncertain or several times apply, `TIME` should not be used" or perhaps with separate explanations for each case.
3. The spec could prohibit `TIME` being used with some or all of these cases.

Other options?

**Comment by dthaler** [comment]
> The issue is that some illogical date/time combinations can be produced but sometimes a logical statement such as in “A child was born in 2012 at 14:32” needs to be recorded.

In my view that would be analogous to saying "A child was born on the 14th in 2012" without mentioning the month.  GEDCOM doesn't support that without using a phrase.  It would seem to me to be odd to allow a specific time but not day if one doesn't allow a specific day but not month.  My preference is that we shouldn't, and just use phrase for such odd things.

**Comment by Norwegian-Sardines** [comment]
> In my view that would be analogous to saying "A child was born on the 14th in 2012" without mentioning the month. GEDCOM doesn't support that without using a phrase.

Are we sure this is the case in practice?  Are you saying that only complete month day year is a valid and all other dates need a phrase?  While I get that this particular date can’t be found on a calendar, can we use a year and a phrase in combination? Previously without a phrase we would have entered the information as written but assumed only the year for timeline display.  This would need to be resolved in migration!

**Comment by ACProctor** [comment]
There appears to be a fundamental fallacy here that every item needs to be digested as a digital "fact" in case some heroic software product can do 
something vague with it.

Missing date fields need a lot more evidential detail than just place holders, for instance. I am not aware of a single product that is going to try 
and sort dates that have missing fields, and some do not even try to sort existing GEDCOM date expressions.

A regex pattern might help but why are software people so scared of putting information/analysis in readable text form; genealogy is performed by 
people, not by software.

If you provide a time value, are we mandating that it has to be UTC (apologies but I can't remember the decision there)? That alone requires the 
genealogist to know the relevant time-zone and its difference from UTC, as well as whether DST was applicable on that historical date. Time zones 
change, and some no longer exist. DST also changes. For example, a WWII time might have been in "British double summer time".

If a date was given as "Christmas before last" then what would you put? Clearly the phrase is interesting, and has to be recorded, but it cannot be 
interpreted without recourse to some background information about the associated source.

In summary, I feel that we should use PHRASE and/or TEXT for such cases. There were much more important issues that were proposed for V7 and yet 
received strong resistance. This sort of low-value change is going to create a rift that GEDCOM doesn't need.

Tony

**Comment by dthaler** [comment]
> Are you saying that only complete month day year is a valid and all other dates need a phrase? 

No.  The spec has:
`date        = [calendar D] [[day D] month D] year [D epoch]`
meaning if you have a `day` then a `month` is required not optional.
I'm saying the analogy is that if you have a `Time` then it would be consistent with that to make the `day` be required.

**Comment by dthaler** [comment]
> I am not aware of a single product that is going to try and sort dates that have missing fields, and some do not even try to sort existing GEDCOM date expressions.

I am.   E.g., sorting dates that only have years.

> If you provide a time value, are we mandating that it has to be UTC (apologies but I can't remember the decision there)? 

GEDCOM 7 supports both UTC and event-local time in TIME payloads.

> If a date was given as "Christmas before last" then what would you put? Clearly the phrase is interesting, and has to be recorded, but it cannot be interpreted without recourse to some background information about the associated source. In summary, I feel that we should use PHRASE and/or TEXT for such cases. 

I agree.

**Comment by ACProctor** [comment]
OK, point taken, Dave. I meant missing year when you have month, or missing month when you have a day, i.e. not the ISO elided forms. :-)

Tony

**Comment by Norwegian-Sardines** [comment]
> If a date was given as "Christmas before last" then what would you put?

If the phrase was “Christmas before my 10th birthday I got a pony.” could you extrapolate the year and day month?  Probably.  I’d try to enter that as (for example) ABT DEC 25 1912.

**Comment by hartenthaler** [comment]
I believe that one source of the problem is that the standard uses 

2 DATE dd mmm yyyy (or any other allowed expression: DatePeriod / dateRange / dateApprox)
3 TIME hh:mm (or any other allowed expression for time)

instead of using

2 DATE hh:mm dd mmm yyyy

ie a definition like date = [calendar D] [[[time] day D] month D] year [D epoch]
Then for me, the semantics are much clearer.

> - Does FROM <date> TO <date> with TIME <time> mean the multi-day event both started and stopped at that time? Or was it a periodic event that took place each day at that time during that period?

=> "FROM 12:34 1 JAN 2000 TO 01:23 31 DEC 2000" has precise starting time/date and ending time/date; no periodic event

> - Does BET <date> AND <date> with TIME <time> mean we're not sure when it happened but we know it was between that time on those two dates? Or does it mean that we don't know the date, but we do know the time? (I think this second option is what both @Norwegian-Sardines and @fisharebest are proposing in their comments)

=> "BET 12:34 1 JAN 2000 AND 01:23 31 DEC 2000" means that we're not sure when it happened but we know it was between the two time/date values; this is maybe useful for a DEAT event when the death certificate says "He died at 1st January 2000 between  01:00 and 04:00 o'clock".

> - Does DATE 2022 with TIME 11:50 mean 11:50 on some unknown day in 2022?

=> this would not be allowed, because if there is time, then you need an exact date

> - Does DATE ABT 24 OCT 2022 with TIME 11:50 Z mean we're unsure of the date but know the time? Or does it mean we are unsure of the exact time but know it was close to 2022-10-24 11:50:00 UTC?

=> "DATE ABT 11:50 Z 24 OCT 2022" means that we are unsure of the exact time but know it was close to 2022-10-24 11:50:00 UTC.

**Comment by tychonievich** [comment]
Discussed 2022-10-25
- For approximate `DATE`s, we should state that `TIME` is not approximate. We should consider adding approximate times in a future version of the spec.
- For date periods (i.e. events that spanned multiple days) we should recommend against adding a `TIME` and say that if it is present its meaning is not defined by the spec.
- In a future version we might propose a new date/time format, probably proposed by a working group on dates, times, and calendars, but such a group is not currently active.

**Comment by hartenthaler** [comment]
Ok, I see that my suggestion would be a major change and should be prepared more carefully than the existing solution by adding TIME one level below DATE. 

@tychonievich : all your three statements would help to make the standard better than it is now.

It is then still possible to describe that a child is born on 1 JAN 2000 at 12:34 o'clock.
1 BIRT
2 DATE 1 JAN 2000
3 TIME 12:34
ok.

How would you store the information "He died on 1st January 2000 between 01:00 and 04:00 o'clock" in GEDCOM? or a bit more complex "He died between 31 December 1999 23:50 o'clock and 1st January 2000 00:10 o'clock"?

**Comment by tychonievich** [comment]
> How would you store the information "He died on 1st January 2000 between 01:00 and 04:00 o'clock" in GEDCOM? or a bit more complex "He died between 31 December 1999 23:50 o'clock and 1st January 2000 00:10 o'clock"?

These do not have machine-parseable solutions in GEDOM 7.0 (nor in 5.5 nor earlier formats). Instead you'd need an approximate date with a `DATE`.`PHRASE` to put these in a human language instead. See also #230 which proposes adding a `TIME`.`PHRASE` in an upcoming version 7.1.

**Comment by tychonievich** [comment]
Discussed in context of clarifying the ambiguity in a patch as part of #241 in steering committee. Two options discussed

Option 1: If there is a `TIME`, that is asserting that the event happened at a specific time on a single day. It is inconsistent to use `TIME` with Date Period but permissible for all other date types.

Option 2: We could pick a subset of `DateValue` productions (`date` with `day` included, possibly also `dateApprox` and `dateRange`) and say TIME should only be used for them.

We acknowledge that none of these are optimal, due to the limitations of date/time in 7.0. We'd like to be able to provide a time for each `date` in a multi-`date` `DateValue` and may add that in 7.1 or 8.0. For patching 7.0, option 1 seems less distruptive so we plan to use it as a patch.
